packages/components/src/internal/components/picklist/PicklistEditModal.tsx

```diff
@@ -65,10 +73,16 @@ export const PicklistEditModal: FC<Props> = memo(props => {
 
     const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
     const [picklistError, setPicklistError] = useState<string>(undefined);
+    const [sampleIds, setSampleIds] = useState<string[]>(initSampleIds);
     const [validCount, setValidCount] = useState<number>(selectedQuantity ? selectedQuantity : sampleIds?.length);
     const [statusData, setStatusData] = useState<OperationConfirmationData>(undefined);
+    const selectionKey = useMemo(
+        //If we are sampleFieldKey is set then selectionKey isn't going to correctly indicate selected samples
+        () => (sampleFieldKey ? undefined : initSelectionKey),
+        [sampleFieldKey, initSelectionKey]
+    );
```

Comment:
Same comments here as for `ChoosePicklistModal` regarding field naming, and lifecycle.